gcloud/logging/handlers/handlers.py

```diff
@@ -70,22 +77,54 @@ class CloudLoggingHandler(logging.StreamHandler, object):
 
     """
 
+    LOG_RECORD_DATA_WHITELIST = {
+        'process', 'processName', 'thread', 'threadName', 'created',
+        'relativeCreated', 'pathname', 'filename', 'lineno', 'module',
+        'funcName', 'levelno'
+    }
+
     def __init__(self, client,
                  name=DEFAULT_LOGGER_NAME,
-                 transport=BackgroundThreadTransport):
+                 transport=BackgroundThreadTransport,
+                 extra_fields=tuple()):
         super(CloudLoggingHandler, self).__init__()
         self.name = name
         self.client = client
+        self.extra_fields = extra_fields
         self.transport = transport(client, name)
 
+    def generate_log_struct_kwargs(self, record, message):
+        """Produces the dictionary of chosen LogRecord fields. extra_fields
+        are taken into account.
+
+        :type record: :class:`logging.LogRecord`
+        :param record: Python log record
+
+        :type message: str
+        :param message: The formatted log message
+
+        :return: dict with keyword arguments for
+                 :method:`gcloud.logging.Logger.log_struct()`.
+        """
+        return {
+            'info': {
+                'message': message,
+                'python_logger': record.name,
+                'data': {k: getattr(record, k, None) for k in itertools.chain(
```

Comment:
for readability, probably better to refactor this above to a temp data variable that you just set here